cpp/velox/shuffle/VeloxSplitter.cc

```diff
@@ -365,115 +364,30 @@ arrow::Status VeloxSplitter::SplitFixedWidthValueBuffer(const velox::RowVector&
   for (auto col = 0; col < fixed_width_column_count_; ++col) {
     auto col_idx = simple_column_indices_[col];
     auto column = rv.childAt(col_idx);
-    assert(column->isFlatEncoding());
-
-    const uint8_t* src_addr = (const uint8_t*)column->valuesAsVoid();
     const auto& dst_addrs = partition_fixed_width_value_addrs_[col];
 
     switch (arrow::bit_width(arrow_column_types_[col_idx]->id())) {
       case 8:
-        RETURN_NOT_OK(SplitFixedType<uint8_t>(src_addr, dst_addrs));
+        RETURN_NOT_OK(SplitFixedType<uint8_t>(column, dst_addrs));
         break;
       case 16:
-        RETURN_NOT_OK(SplitFixedType<uint16_t>(src_addr, dst_addrs));
+        RETURN_NOT_OK(SplitFixedType<uint16_t>(column, dst_addrs));
         break;
       case 32:
-        RETURN_NOT_OK(SplitFixedType<uint32_t>(src_addr, dst_addrs));
+        RETURN_NOT_OK(SplitFixedType<uint32_t>(column, dst_addrs));
         break;
       case 64:
-#ifdef PROCESSAVX
```

Comment:
When we will define `PROCESSAVX`? @FelixYBW How can I test performance difference with `PROCESSAVX`

Reply:
I found that the macro `PROCESSAVX ` was not defined in `splitter.cc` also, so I remove it now.
I think that it only affects the performance, not the correctness, We can test it in the future.